packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -0,0 +1,278 @@
+   * SSH key but do not specify a source security group when you create a managed node group, then port 22 on the worker nodes is
+   * opened to the internet (0.0.0.0/0)
+   *
+   * @default - disabled
+   */
+  readonly remoteAccess?: NodegroupRemoteAccess;
+  /**
+   * The metadata to apply to the node group to assist with categorization and organization. Each tag consists of a key and an optional
+   * value, both of which you define. Node group tags do not propagate to any other resources associated with the node group, such as the
+   * Amazon EC2 instances or subnets.
+   *
+   * @default - None
+   */
+  readonly tags?: { [name: string]: string };
+}
+
+/**
+ * NodeGroup properties interface
+ */
+export interface NodegroupProps extends NodegroupOptions {
+  /**
+   * Cluster resource
+   * [disable-awslint:ref-via-interface]"
+   */
+  readonly cluster: Cluster;
+}
+
+/**
+ * The Nodegroup resource class
+ */
+export class Nodegroup extends Resource implements INodegroup {
+  /**
+   * Import the Nodegroup from attributes
+   */
+  public static fromNodegroupName(scope: Construct, id: string, nodegroupName: string): INodegroup {
+    class Import extends Resource implements INodegroup {
+      public readonly nodegroupName = nodegroupName;
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * ARN of the nodegroup
+   *
+   * @attribute
+   */
+  public readonly nodegroupArn: string;
+  /**
+   * Nodegroup name
+   *
+   * @attribute
+   */
+  public readonly nodegroupName: string;
+  /**
+   * the Amazon EKS cluster resource
+   *
+   * @attribute ClusterName
+   */
+  public readonly cluster: Cluster;
+  /**
+   * IAM role of the instance profile for the nodegroup
+   */
+  public readonly role: IRole;
+
+  private readonly desiredSize: number;
+  private readonly maxSize: number;
+  private readonly minSize: number;
+
+  constructor(scope: Construct, id: string, props: NodegroupProps ) {
+    super(scope, id);
+
+    this.cluster = props.cluster;
+
+    this.desiredSize = props.desiredSize ?? props.minSize ?? 2;
+    this.maxSize = props.maxSize ?? this.desiredSize;
+    this.minSize = props.minSize ?? 1;
+
+    if (this.desiredSize > this.maxSize) {
+      throw new Error(`Desired capacity ${this.desiredSize} can't be greater than max size ${this.maxSize}`);
+    }
+    if (this.desiredSize < this.minSize) {
+      throw new Error(`Minimum capacity ${this.minSize} can't be greater than desired size ${this.desiredSize}`);
+    }
+
+    const ngRole = new Role(this, 'NodeGroupRole', {
+      assumedBy: new ServicePrincipal('ec2.amazonaws.com')
+    });
+
+    ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'));
+    ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'));
+    ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'));
+
+    const resource = new CfnNodegroup(this, id, {
```

Comment:
Instead of “id” this should be “Resource”. That’s our convention and will enable users to use “findDefaultChild”